src/cli.rs

```diff
@@ -10,25 +10,23 @@ use crate::tap;
 use crate::top;
 use crate::{config, generate, get_version, graph, list, unit_test, validate};
 
-#[derive(StructOpt, Debug)]
-#[structopt(rename_all = "kebab-case")]
+#[derive(Parser, Debug)]
+#[clap(rename_all = "kebab-case")]
 pub struct Opts {
-    #[structopt(flatten)]
+    #[clap(flatten)]
     pub root: RootOpts,
 
-    #[structopt(subcommand)]
+    #[clap(subcommand)]
     pub sub_command: Option<SubCommand>,
 }
 
 impl Opts {
     pub fn get_matches() -> Self {
         let version = get_version();
-        let app = Opts::clap().version(version.as_str()).global_settings(&[
-            AppSettings::ColoredHelp,
-            AppSettings::InferSubcommands,
-            AppSettings::DeriveDisplayOrder,
-        ]);
-        Opts::from_clap(&app.get_matches())
+        let app = Opts::command()
+            .version(version.as_str())
+            .global_setting(AppSettings::DeriveDisplayOrder);
```

Comment:
`AppSettings::ColoredHelp` is a default in clap 3, so this was removed.

 `AppSettings::InferSubcommands` has been deprecated in favour of https://docs.rs/clap/latest/clap/struct.App.html#method.infer_subcommands

It briefly crossed my mind that this may be a culprit for the collision between the `-c` flag and `config` subcommand in https://github.com/vectordotdev/vector/pull/11442, although they should be distinct domains, so may be nothing.

In any case, I don't know how useful (or encouraged) it is to type, say, `vector vr` instead of `vector vrl`. I'd prefer explicit names personally given the growing number of subcommands.

Again, strong opinions welcome.

Reply:
Yeah, agreed, I like the full command names.